packages/frontend/src/utils/data-decoding.ts

```diff
@@ -1,40 +1,52 @@
 import { Amount, Fetcher } from "@carrot-kpi/sdk";
-import { Provider } from "@wagmi/core";
-import { BigNumber } from "ethers";
-import { defaultAbiCoder } from "ethers/lib/utils.js";
+import { type PublicClient } from "wagmi";
 import { CollateralData } from "../creation-form/types";
+import { Hex, decodeAbiParameters } from "viem";
```

Comment:
`Hex` can be imported as type.

Reply:
Addressed in f0c94f69958942158d917d53ea5549df263f0c66